carapace-server/src/main/java/org/carapaceproxy/server/mapper/StandardEndpointMapper.java

```diff
@@ -414,7 +414,7 @@ public MapResult map(HttpRequest request, String userId, String sessionId, Backe
                     }
 
                     BackendConfiguration backend = this.backends.get(backendId);
-                    if (backend != null && backendHealthManager.isAvailable(backendId)) {
+                    if (backend != null && backendHealthManager.isAvailable(backend.getHostPort())) {
```

Comment:
this fix seems unrelated to the title

Reply:
Whenever a backend availability is checked a new BackendHealthStatus object is created. This is because the id is always new due to backends list resetting performed in BackendHealthManager.run according to the configuration (which consider host:port as id). The conseguences is that the new BackendHealthStatus with default reportedAsUnreachable = false  seems to be always consider as available.